compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -119,6 +119,10 @@ impl<'ll, 'tcx> IntrinsicCallMethods<'tcx> for Builder<'_, 'll, 'tcx> {
             sym::likely => {
                 self.call_intrinsic("llvm.expect.i1", &[args[0].immediate(), self.const_bool(true)])
             }
+            sym::is_val_statically_known => self.call_intrinsic(
+                &format!("llvm.is.constant.{:?}", args[0].layout.llvm_type(self.cx)),
```

Comment:
maybe we need to use `immediate_llvm_type` here?

```suggestion
                &format!("llvm.is.constant.{:?}", args[0].layout.immediate_llvm_type(self.cx)),
```

Reply:
Hmm, what's the difference between the two? I don't know much about LLVM, hence the long wait for getting CI to pass...

Reply:
llvm_type is the type used to pass arguments of that Rust type between functions; immediate_llvm_type is the type used for LLVM SSA values of that Rust type.